### PR TITLE
Pull Request for Issue2037: generalizing BioXASBeamStatus

### PR DESCRIPTION
**Pull request by helfrij**
- Modified BioXASBeamStatus, it no longer has specific setters for particular beamline components. Instead, components can be added as needed. 
- As only one state needs to be specified for the beam status (ie. the beam on value), this control is truly a bi-state control, whereas other controls that inherit from BioXASBiStateGroup (ie BioXASShutters, BioXASValves, etc) are really more like 'tri-state' controls. They need two values, an open and a closed value (plus implicit between/unknown). Created new class BioXASTriStateControl for this purpose, that inherits from and extends BioXASBiStateControl. Not sure if this is a good way to go long-term.
- Modified the BioXASBeamStatusBar in response to the changes in BioXASBeamStatus.
- Encountered problem with the beam status, where it would report as 'not connected' even though it had an updateConnected() implementation in an ancestor class. Since these reimplementations are identical to the one now in AMPseudoMotorControl by default, I removed them. This seems to have fixed the problem.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @iainworkman, would you take a look at this request?

File: source/beamline/BioXAS/BioXASBiStateGroup.cpp [hunk 1]
```diff
@@ -107,31 +107,31 @@ bool BioXASBiStateGroup::canStop() const
 	return result;
 }
 
-void BioXASBiStateGroup::updateConnected()
-{
-	bool connected = false;
+//void BioXASBiStateGroup::updateConnected()
```
Comment (iainworkman) [inline source/beamline/BioXAS/BioXASBiStateGroup.cpp]: Do you wish to leave this commented code in place, or can it be removed?

File: source/beamline/BioXAS/BioXASBiStateGroup.h [hunk 2]
```diff
@@ -21,13 +21,13 @@ class BioXASBiStateGroup : public AMEnumeratedControl
 	virtual bool canStop() const;
 
 protected slots:
-	/// Updates the connected state.
-	virtual void updateConnected();
+//	/// Updates the connected state.
+//	virtual void updateConnected();
```
Comment (iainworkman) [inline source/beamline/BioXAS/BioXASBiStateGroup.h]: Can this commented code be removed?

File: source/ui/BioXAS/BioXASControlEditor.cpp [hunk 3]
```diff
@@ -407,9 +407,14 @@ AMNumber BioXASControlEditor::getCalibratedDoubleValue()
 
 	return result;
 }
-
+#include <QDebug>
 void BioXASControlEditor::onContextMenuRequested(const QPoint &clickPosition)
 {
+	if (control_) {
+		qDebug() << "\n\n";
```
Comment (iainworkman) [inline source/ui/BioXAS/BioXASControlEditor.cpp]: Can these debugs be removed?

**Comment by iainworkman** [comment]
Looks good to me.
Time Estimate: 15 minutes.

File: source/beamline/BioXAS/BioXASBiStateGroup.cpp (hunk 1 above)
Comment (helfrij) [inline source/beamline/BioXAS/BioXASBiStateGroup.cpp]: I'll remove it.

File: source/beamline/BioXAS/BioXASBiStateGroup.h (hunk 2 above)
Comment (helfrij) [inline source/beamline/BioXAS/BioXASBiStateGroup.h]: Will remove this too.

File: source/ui/BioXAS/BioXASControlEditor.cpp (hunk 3 above)
Comment (helfrij) [inline source/ui/BioXAS/BioXASControlEditor.cpp]: Yup, these are now removed too.

**Comment by dretrex** [comment]
Looks good.
Time Estimate: 5 minutes

**Pull request merged**